src/core/pointcloud/qgspointcloudlayer.cpp

```diff
@@ -839,11 +858,35 @@ void QgsPointCloudLayer::resetRenderer()
   {
     calculateStatistics();
   }
-  if ( mRenderer->type() == QLatin1String( "extent" ) )
+  if ( !mRenderer || mRenderer->type() == QLatin1String( "extent" ) )
   {
     setRenderer( QgsPointCloudRendererRegistry::defaultRenderer( this ) );
   }
   triggerRepaint();
 
   emit rendererChanged();
 }
+
+void QgsPointCloudLayer::saveCopcStatistics()
+{
+  // If the point cloud doesn't have a valide PDAL data provider and a valid index we don't save any statistics
+  if ( !mDataProvider || !mDataProvider->index() || !mDataProvider->index()->isValid() || mDataProvider->name() != QStringLiteral( "pdal" ) )
+    return;
+  if ( QgsCopcPointCloudIndex *index = qobject_cast<QgsCopcPointCloudIndex *>( mDataProvider->index() ) )
+  {
+    index->writeStats( mStatistics );
+  }
+}
+
+QgsPointCloudStatistics QgsPointCloudLayer::loadCopcStatistics()
+{
+  // If the point cloud doesn't have a valide PDAL data provider and a valid index we don't load anything and return an empty statistics object
+  if ( !mDataProvider || !mDataProvider->index() || !mDataProvider->index()->isValid() || mDataProvider->name() != QStringLiteral( "pdal" ) )
+    return QgsPointCloudStatistics();
+
+  if ( QgsCopcPointCloudIndex *index = qobject_cast<QgsCopcPointCloudIndex *>( mDataProvider->index() ) )
+  {
+    return index->readStats();
+  }
```

Comment:
Is this going to build if compiled with WITH_COPC=FALSE ?

It is generally safer to add this kind of methods as virtual methods and only relevant providers would contain actual implementation.

Reply:
I removed these 2 function and replaced the call with the if () { save/load } and added ifdefs around that